File: Python/백준/Bronze/27325. 3 つの箱 （Three Boxes）/3 つの箱 （Three Boxes）.py
a = int(input())
cnt = 0
x = 1
aa = input()
for i in aa:
    if i=='L' and x!=1:
        x-=1
    elif i=='R' and x!=1:
        cnt+=1
        x=3
    elif i=='R' and x==1:
        x+=1
print(cnt)

File: Java/백준/Bronze/27325. 3 つの箱 （Three Boxes）/3 つの箱 （Three Boxes）.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int a = in.nextInt();
        int x = 1;
        int cnt = 0;
        String s = in.next();
        for (int i=0;i<a;i++){
            char c = s.charAt(i);
            if(c=='L'){
                if(x!=1){x--;}
            }
            else{
                if(x!=3){x++;}
                if(x==3){cnt++;}
            }
        }
        System.out.println(cnt);
    }
}